src/georaster.d.ts

```diff
@@ -0,0 +1,83 @@
+declare function parseGeoraster(
+  data: object | string | Buffer | ArrayBuffer | number[][][],
+  /** the raster metadata */
+  metadata?: parseGeoraster.GeorasterMetadata,
+  /** whether or not to print debug statements */
+  debug?: boolean
+): Promise<parseGeoraster.Georaster>;
+
+// Match default CJS export in index.js
+export = parseGeoraster;
+
+// A namespace with the same name as the default export is needed to define additional type exports
+// https://stackoverflow.com/a/51238234/4159809
+declare namespace parseGeoraster {
+  /** defines the new raster image to generate as a window in the source raster image.  Resolution (cell size) is determined from this */
+  export interface WindowOptions {
+    /** left side of the image window in pixel coordinates */
+    left: number
+    /** top of the image window in pixel coordinates */
+    top: number
+    /** right of the image window in pixel coordinates.  Should be greater than left */
+    right: number
+    /** bottom of the image window in pixel coordinates.  Should be greater than top */
+    bottom: number
+    /** width in pixels to make the resulting raster.  Will resample and/or use overview if not same as right - left */
+    width: number
+    /** height in pixels to make the resulting raster.  Will resample and/or use overview if not same as bottom - top */
+    height: number
+    /** method to map src raster values to result raster. Supports 'nearest' neighbor, defaults to 'bilinear' */
+    resampleMethod?: string
+  }
+  
+  export interface Georaster {
+    /** raster values.  first dimension is raster band, remainder is 2D array of cell values */
+    values: number[][][];
```

Comment:
In practice, I've found the raster values returned are not always of type floating point `number`.  If your raster is 8-bit unsigned integers for example (aka of type Byte in GDAL), then a Uint8Array of values will be returned.  You don't access the values any differently I've found so far, so this PR could land and be a good start, it's just not completely accurate from a type perspective.

I'll follow-on with testing out using something like the `TypedArray` type drafted for working with geotiff.js over here - https://github.com/hms-dbmi/viv/blob/53478495580090ccd6f86f33dd350dbb0806a79f/src/missing-types.d.ts#L38.

```
/** 2D array of pixel data */
type DataArray =
  | number[]
  | Uint8Array
  | Int8Array
  | Uint16Array
  | Int16Array
  | Uint32Array
  | Int32Array
  | Float32Array
  | Float64Array;
  
/**
 * Raster data consists of one or more bands, with a 2D array of pixel values
 * The pixel data type will vary by raster */
type RasterData = DataArray[]
```

Really these base types should be provided by geotiff.js and I can look to contribute that upstream.  Then these georaster and geoblaze types downstream can make use of them.  One step at a time.

Reply:
I have added a `TypedArray` to this PR that cover this.